TRAMS-API/Startup.cs

```diff
@@ -114,30 +116,44 @@ private static void ConfigureRepositories(IServiceCollection services)
+                                  GetAcademiesResponseMapper>();
 
-            services.AddTransient<IMapper<PostProjectsRequestModel, PostAcademyTransfersProjectsD365Model>>(r =>
-                new PostProjectsModelToPostProjectsD365ModelMapper());
+            services.AddTransient<IMapper<PostProjectsRequestModel, PostAcademyTransfersProjectsD365Model>, 
+                                  PostProjectsRequestMapper>();
+
+            services.AddTransient<IMapper<AcademyTransfersProjectAcademy, GetProjectsAcademyResponseModel>, 
+                                  GetProjectAcademiesResponseMapper>();
+
+            services.AddTransient<IMapper<GetProjectsD365Model, GetProjectsResponseModel>,
+                                  GetProjectsResponseMapper>();
+            
         }
 
         private static void ConfigureHelpers(IServiceCollection services)
         {
-            services.AddTransient<ID365ModelHelper<GetTrustsD365Model>>(r =>
-                new D365ModelHelper<GetTrustsD365Model>());
-
-            services.AddTransient<ID365ModelHelper<GetAcademiesD365Model>>(r =>
-               new D365ModelHelper<GetAcademiesD365Model>());
-
-            services.AddTransient<ID365ModelHelper<GetAcademyTransfersProjectsD365Model>>(r =>
-                new D365ModelHelper<GetAcademyTransfersProjectsD365Model>());
-
-            services.AddTransient<IOdataUrlBuilder<GetTrustsD365Model>, ODataUrlBuilder<GetTrustsD365Model>>();
-            services.AddTransient<IOdataUrlBuilder<GetAcademiesD365Model>, ODataUrlBuilder<GetAcademiesD365Model>>();
-            services.AddTransient<IOdataUrlBuilder<GetAcademyTransfersProjectsD365Model>, ODataUrlBuilder<GetAcademyTransfersProjectsD365Model>>();
+            services.AddTransient<ID365ModelHelper<GetTrustsD365Model>,
+                                  D365ModelHelper<GetTrustsD365Model>>();
+            services.AddTransient<ID365ModelHelper<GetAcademiesD365Model>, 
+                                  D365ModelHelper<GetAcademiesD365Model>>();
+            services.AddTransient<ID365ModelHelper<GetProjectsD365Model>,
+                                  D365ModelHelper<GetProjectsD365Model>>();
+            services.AddTransient<ID365ModelHelper<AcademyTransfersProjectAcademy>, 
+                                  D365ModelHelper<AcademyTransfersProjectAcademy>>();
+
+            services.AddTransient<IOdataUrlBuilder<GetTrustsD365Model>, 
+                                  ODataUrlBuilder<GetTrustsD365Model>>();
+            services.AddTransient<IOdataUrlBuilder<GetAcademiesD365Model>, 
+                                  ODataUrlBuilder<GetAcademiesD365Model>>();
+            services.AddTransient<IOdataUrlBuilder<GetProjectsD365Model>, 
+                                  ODataUrlBuilder<GetProjectsD365Model>>();
+            services.AddTransient<IOdataUrlBuilder<AcademyTransfersProjectAcademy>, 
+                                  ODataUrlBuilder<AcademyTransfersProjectAcademy>>();
```

Comment:
Suggestion:  Probably overkill for now but other IoC frameworks tend to have methods that automatically map all instances of a set type using a naming convention.  For example I know with StructureMap in the past I've been able to say something like add all types conforming to IOdataUrlBuilder<>, and it would map them based on a convention.  This way you don't have to explicitly call out each one every time a new instance is added to the application.  As a low priority it might be worth exploring if the .NET Core container has something similar.

Alternatively we could create our own helper method which finds all types which close an interface and registers them.

Reply:
I didn't know you can do it but this works. Good shout!
services.AddTransient(typeof(ID365ModelHelper<>), typeof(D365ModelHelper<>));
services.AddTransient(typeof(IOdataUrlBuilder<>), typeof(ODataUrlBuilder<>));